plugins/woocommerce/src/Blocks/BlockTypes/Checkout.php

```diff
@@ -453,6 +453,48 @@ protected function enqueue_data( array $attributes = [] ) {
 		$this->asset_data_registry->add( 'collectableMethodIds', $local_pickup_method_ids );
 		$this->asset_data_registry->add( 'shippingMethodsExist', CartCheckoutUtils::shipping_methods_exist() > 0 );
 
+		/**
+		 * This section handles the shipping address fields that trigger shipping rate recalculation.
+		 * The address_fields_for_shipping_rates array contains the default fields that affect shipping rates.
+		 * Additional fields names can be added via the 'woocommerce_essential_shipping_data' filter.
+		 * The filtered array is validated to ensure all entries are strings before being registered.
+		 */
+
+		$address_fields_for_shipping_rates = [ 'state', 'country', 'postcode', 'city' ];
```

Comment:
After this PR is merged I'll make a separate PR for the `isAddressComplete` logic to also account for these filtarable values [here](https://github.com/woocommerce/woocommerce/blob/bae680d073e9650545e113dd8b87098f9ad566ed/plugins/woocommerce/client/blocks/assets/js/base/utils/address.ts#L50-L53) JS & [here](https://github.com/woocommerce/woocommerce/pull/53466/files#diff-7ac6bf2178d0dd835385fdca2ccde805a39189d75c90d438e9a27a758084470cR45-R50) in PHP,